zero-loads-app/src/loads/api/api.py

```diff
@@ -23,43 +29,66 @@
 async def lifespan(app: FastAPI):
     """Function that handles startup and shutdown events (https://fastapi.tiangolo.com/advanced/events/)"""
     sessionmanager.initialize(settings.pg_url)
-    yield
-    if sessionmanager._engine is not None:
-        await sessionmanager.close()
+    async with MqttClient(settings.mqtt_host, settings.mqtt_port) as mqtt:
+        messaging = Messaging(mqtt_client=mqtt, modules=[sail_systems], variable_definition=loads_variables)
+        run_task = create_task(await messaging.run())
+
+        def _finish(task: Task):
+            if err := task.exception():
+                logger.critical("Messaging failed", exc_info=err)
+                sys.exit(1)
+
+        run_task.add_done_callback(_finish)
+        app.state.messaging = messaging
+
+        yield
+        run_task.cancel()
+        if sessionmanager._engine is not None:
+            await sessionmanager.close()
 
 
 async def get_db_session():
     async with sessionmanager.session() as session:
         yield session
 
 
+def get_messaging() -> Messaging:
+    return app.state.messaging
+
+
 @dataclass
-class MyContext(BaseContext):
+class LoadsContext(BaseContext):
+    messaging: Messaging
     session: AsyncSession
 
 
 async def get_context(
+    messaging: Annotated[Messaging, Depends(get_messaging)],
     session: Annotated[AsyncSession, Depends(get_db_session)],
-) -> MyContext:
-    return MyContext(session)
+) -> LoadsContext:
+    return LoadsContext(messaging=messaging, session=session)
 
 
 @strawberry.type
 class Query:
```

Comment:
I thought we'd agreed an a `variables` containing `actual` and `reference`. Sure you could also have them as top level (then as plurals), but not exclusively.